Dockerfile.alpine

```diff
@@ -0,0 +1,42 @@
+FROM python:3.7.3-alpine3.9 AS build
+
+ARG DOCKER_COMPOSE_GITSHA=unknown
+ENV DOCKER_COMPOSE_GITSHA=$DOCKER_COMPOSE_GITSHA
+
+ENV BUILD_BOOTLOADER=1
+
+RUN apk add --no-cache \
+    bash \
+    build-base \
+    ca-certificates \
+    curl \
+    gcc \
+    git \
+    libc-dev \
+    libffi-dev \
+    libgcc \
+    make \
+    musl-dev \
+    openssl \
+    openssl-dev \
+    python2 \
+    python2-dev \
+    zlib-dev && \
+    pip install virtualenv==16.2.0 tox==2.1.1
+# FIXME(chris-crone): virtualenv 16.3.0 breaks build, force 16.2.0 until fixed
+
+WORKDIR /code/
+
+ADD . /code/
+RUN tox --notest
+RUN script/build/linux-entrypoint
+
+ADD docker-compose-entrypoint.sh /usr/local/bin/
+COPY --from=docker:18.09.4 /usr/local/bin/docker /usr/local/bin/docker
+ENTRYPOINT ["sh", "/usr/local/bin/docker-compose-entrypoint.sh"]
+
+FROM alpine:3.9 AS runtime
+COPY --from=docker:18.09.4 /usr/local/bin/docker /usr/local/bin/docker
+COPY --from=build /usr/local/bin/docker-compose /usr/local/bin/docker-compose
+ADD docker-compose-entrypoint.sh /usr/local/bin/
+ENTRYPOINT ["sh", "/usr/local/bin/docker-compose-entrypoint.sh"]
```

Comment:
Tiny optimisation when using "classic" builder; this step can be done _before_ the `COPY` lines (although BuildKit will already be smart enough to cache it)

Reply:
In this case, I think that what we win is not worth losing the readability of having the `ENTRYPOINT` as last line of a target. 
@thaJeztah WDYT?